### PR TITLE
collapse printing paths

### PR DESCRIPTION
**Pull request by deads2k**
This collapses our printers into:

1. `PrintSuccess` - which was special cased before and should really just be a flavor of printer
2. `PrintObject` - prints an object using all the defaults.  This should go away once we have options on all our commands.
3. `PrinterForOptions` - which gives back a printer based on the options.  This should really be a method on the options themselves.

And none of these methods have any factory dependencies anymore.

@kubernetes/sig-cli-maintainers 
@soltysh @juanvallejo  You two have followed the work.  Let's tie this off.

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/cmd_test.go [hunk 1]
```diff
@@ -197,12 +197,6 @@ func stringBody(body string) io.ReadCloser {
 
 func Example_printReplicationControllerWithNamespace() {
```
Comment (liggitt) [inline pkg/kubectl/cmd/cmd_test.go]: Lost namespace option here, which was the point of this test

File: pkg/kubectl/cmd/cmd_test.go [hunk 2]
```diff
@@ -590,17 +557,11 @@ func Example_printPodShowAll() {
 
 func Example_printServiceWithNamespacesAndLabels() {
 	f, tf, _, ns := cmdtesting.NewAPIFactory()
-	p := printers.NewHumanReadablePrinter(nil, nil, printers.PrintOptions{
-		WithNamespace: true,
```
Comment (liggitt) [inline pkg/kubectl/cmd/cmd_test.go]: Lost namespace option here

File: pkg/kubectl/cmd/delete_test.go [hunk 3]
```diff
@@ -307,10 +302,10 @@ func TestDeleteObjectGraceZero(t *testing.T) {
 	cmd := NewCmdDelete(fake, buf, errBuf)
 	cmd.Flags().Set("output", "name")
 	cmd.Flags().Set("grace-period", "0")
-	cmd.Run(cmd, []string{"pod/nginx"})
+	cmd.Run(cmd, []string{"pods/nginx"})
```
Comment (liggitt) [inline pkg/kubectl/cmd/delete_test.go]: Was this change required?

File: pkg/kubectl/cmd/resource/get_test.go [hunk 4]
```diff
@@ -246,7 +244,8 @@ func TestGetUnknownSchemaObject(t *testing.T) {
 	cmd.Run(cmd, []string{"type", "foo"})
 
 	expected := []runtime.Object{cmdtesting.NewInternalType("", "", "foo")}
-	actual := tf.Printer.(*testPrinter).Objects
+	actual := []runtime.Object{}
+	//actual := tf.Printer.(*testPrinter).Objects
```
Comment (liggitt) [inline pkg/kubectl/cmd/resource/get_test.go]: Unused, remove?

File: pkg/kubectl/cmd/util/printing.go [hunk 5]
```diff
@@ -82,10 +85,54 @@ func ValidateOutputArgs(cmd *cobra.Command) error {
 	return nil
 }
 
-// printerForOptions returns the printer for the outputOptions (if given) or
-// returns the default printer for the command. Requires that printer flags have
-// been added to cmd (see AddPrinterFlags).
-func printerForOptions(options *printers.PrintOptions) (printers.ResourcePrinter, error) {
+// PrintSuccess prints a success message and can do a "-o name" as "shortOutput"
+// TODO this should really just a printer
```
Comment (liggitt) [inline pkg/kubectl/cmd/util/printing.go]: ?

Comment (soltysh) [reply pkg/kubectl/cmd/util/printing.go]: I think this TODO goes along the other one, when we have `PrintOptions` it'll be handled by a single printer.

**Review by soltysh — COMMENTED** [review]
This lgtm, except for the namespace comments from Jordan, which needs addressing.

File: pkg/kubectl/cmd/cmd_test.go (hunk 1 above)
Comment (deads2k) [reply pkg/kubectl/cmd/cmd_test.go]: > Lost namespace option here, which was the point of this test

The test faked a printer in a way the command couldn't run and then tested their fake printer.  The test was dumb.  It's either this is or removal.  I don't care which.

File: pkg/kubectl/cmd/cmd_test.go (hunk 2 above)
Comment (deads2k) [reply pkg/kubectl/cmd/cmd_test.go]: > Lost namespace option here

The test faked a printer in a way the command couldn't run and then tested their fake printer. The test was dumb. It's either this is or removal. I don't care which.

File: pkg/kubectl/cmd/delete_test.go (hunk 3 above)
Comment (deads2k) [reply pkg/kubectl/cmd/delete_test.go]: > Was this change required?

fake mapper, fake typer, fake client.  One of them was unhappy.

File: pkg/kubectl/cmd/resource/get_test.go (hunk 4 above)
Comment (deads2k) [reply pkg/kubectl/cmd/resource/get_test.go]: > Unused, remove?

History in case someone wonders how it "worked" before even though the test is flawed.

File: pkg/kubectl/cmd/util/printing.go (hunk 5 above)
Comment (deads2k) [reply pkg/kubectl/cmd/util/printing.go]: > ?

Updated.  Printing success for an object is just a particular kind of "print my object" which is exactly what our Printer signatures are for.  This method should die and be replaced by a proper "successprinter"

**Comment by deads2k** [comment]
I deleted the methods that were only testing mocked out printers that can't actually be built by commands and got comments here.

**Review by soltysh — APPROVED** [review]
/lgtm
